app/scripts/contentscript.js

```diff
@@ -7,81 +7,133 @@ import badge from 'project-badge/dist/badge.js';
-const url = 'https://hubcare.ml/hubcare_indicators' + windowurl + '/' 
-console.log("url = " + url)
-console.log('my windowurl = ' + windowurl)
+const repoName = window.location.pathname
 
-//This var create a span
-var hubcareButton = document.createElement('div')
-//This var create a div
-var myprogress = document.createElement('div')
-var node = document.createElement('div')
 var content = saveClass("new-discussion-timeline experiment-repo-nav")
 var repoContent = saveClass("repository-content")
-var reponav =saveClass('reponav js-repo-nav js-sidenav-container-pjax container zh-attached')
-var repoProjects = saveClass('js-selected-navigation-item reponav-item')
+let metrics = []
 
+/**
+ * Return url to hubcare api
+ * @param {string} repoName 
+ */
+const getApiUrl = (repoName) =>
+    `https://hubcare.ml/hubcare_indicators${repoName}/`;
 
+function createBadge(text, progress, id){
+    var myBadge = new badge.Progress({
+        text: text,
+        progress: progress
+    });
+    document.getElementById(id).appendChild(myBadge.asDOMNode())
+}
 
-node.innerHTML = badges()
-hubcareButton.innerHTML = button()
-//Adding the loading div inside the created div is being added
-myprogress.innerHTML = loading()
-// node.innerHTML = graph()
-content[0].insertBefore(node, repoContent[0])
-content[0].insertBefore(myprogress, repoContent[0])
-reponav[0].appendChild(hubcareButton)
-//content[0].insertBefore(hubcareButton, repoContent[0])
+function saveClass(name_class){
+    var element = document.getElementsByClassName(name_class)
+    return element
+}
 
-// var myChart = echarts.init(document.getElementById('my-graph'))
-// var option = {
-//     xAxis: {
-//         type: 'category',
-//         data: ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']
-//     },
-//     yAxis: {
-//         type: 'value'
-//     },
-//     series: [{
-//         data: [820, 932, 901, 934, 1290, 1330, 1320],
-//         type: 'line'
-//     }]
-// }
+function cleanPageContent(){
+    var element = document.getElementsByClassName('repository-content ')
+    element[0].parentNode.removeChild(element[0])
+}
 
-window.addEventListener("load", function load(event){
-    http.open('GET',url)
-    http.send()
-})
+function createCommitChart(){
+    var content = document.getElementsByClassName("new-discussion-timeline experiment-repo-nav")
+    var repoContent = document.getElementsByClassName("repository-content")
+    var node = document.createElement('div')
+    node.innerHTML = graph()
+    content[0].insertBefore(node, repoContent[0])
+    var myChart = echarts.init(document.getElementById('my-graph'))
+    var option = {
+        xAxis: {
+            type: 'category',
+            data: ['Week 1', 'Week 2', 'Week 3', 'Week 4', 'Week 5', 'Week 6', 'Week 7','Week 8','Week 9','Week 10']
+        },
+        yAxis: {
+            type: 'value'
+        },
+        series: [{
+            data: [23, 32, 12, 15, 8, 45, 30, 28, 9, 35],
```

Comment:
Why is this hardcoded?

#35 asks for visualization of real data.

Reply:
But you still can not request the API data